openmetadata-ui/src/main/resources/ui/src/pages/explore/ExplorePage.component.tsx

```diff
@@ -11,146 +11,255 @@
-    });
-  };
+    return tabInfo[0] as ExploreSearchIndex;
+  }, [tab]);
 
-  const fetchEntityCount = async (indexType: SearchIndex) => {
-    const entityType = getEntityTypeByIndex(indexType);
-    try {
-      const { data } = await searchData(
-        searchText,
-        0,
-        0,
-        getFilterString(initialFilter),
-        emptyValue,
-        emptyValue,
-        indexType,
-        showDeleted,
-        true
-      );
-      const count = getTotalEntityCountByType(
-        data.aggregations?.['sterms#EntityType']?.buckets as Bucket[]
-      );
-
-      setTabCounts((prev) => ({ ...prev, [entityType]: count }));
-    } catch (_error) {
-      // eslint-disable-next-line no-console
-      console.error(_error);
+  useEffect(() => {
+    handleSearchIndexChange(searchIndex);
+  }, [searchIndex]);
+
+  const page = useMemo(() => {
+    const pageParam = parsedSearch.page;
+    if (!isString(pageParam) || isNaN(Number.parseInt(pageParam))) {
+      return 1;
     }
-  };
 
-  const fetchCounts = () => {
-    fetchEntityCount(SearchIndex.TABLE);
+    return Number.parseInt(pageParam);
+  }, [parsedSearch.page]);
+
+  useEffect(() => {
+    handlePageChange(page);
+  }, [page]);
+
+  const [searchResults, setSearchResults] =
+    useState<SearchResponse<ExploreSearchIndex>>();
+
+  const [advancesSearchQueryFilter, setAdvancedSearchQueryFilter] =
+    useState<Record<string, unknown>>();
 
-    fetchEntityCount(SearchIndex.TOPIC);
+  const [sortValue, setSortValue] = useState<string>(INITIAL_SORT_FIELD);
 
-    fetchEntityCount(SearchIndex.DASHBOARD);
+  const [sortOrder, setSortOrder] = useState<string>(INITIAL_SORT_ORDER);
 
-    fetchEntityCount(SearchIndex.PIPELINE);
+  const [searchHitCounts, setSearchHitCounts] = useState<SearchHitCounts>();
+
+  const [showDeleted, setShowDeleted] = useState(false);
 
-    fetchEntityCount(SearchIndex.MLMODEL);
-  };
+  const [isLoading, setIsLoading] = useState(true);
 
   useEffect(() => {
-    fetchCounts();
-  }, [searchText, showDeleted, initialFilter]);
+    setIsLoading(true);
+    Promise.all([
+      searchQuery({
+        query: searchQueryParam,
+        searchIndex,
+        queryFilter: advancesSearchQueryFilter,
+        postFilter: elasticsearchPostFilterQuery,
+        sortField: sortValue,
+        sortOrder,
+        pageNumber: page,
+        pageSize: PAGE_SIZE,
+      })
+        .then((res) => res)
+        .then((res) => setSearchResults(res)),
+      Promise.all(
+        [
+          SearchIndex.TABLE,
+          SearchIndex.TOPIC,
+          SearchIndex.DASHBOARD,
+          SearchIndex.PIPELINE,
+          SearchIndex.MLMODEL,
+        ].map((index) =>
+          searchQuery({
+            query: searchQueryParam,
+            pageNumber: 0,
+            pageSize: 0,
+            postFilter: elasticsearchPostFilterQuery,
+            queryFilter: advancesSearchQueryFilter,
+            searchIndex: index,
+            // includeDeleted: showDeleted,
```

Comment:
Can we remove this?
This feature will be disable for now or how did you plan it?

Reply:
Will remove. 

The serachAPI backend still supports an `includeDeleted` flag for backward compatibility, but I plan to deprecate that since `deleted` can be done like another filter, it doesn't need preferential treatment.